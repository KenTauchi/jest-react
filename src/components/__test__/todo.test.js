import { render, screen, cleanup } from "@testing-library/react";
import Todo from "../todo";
import renderer from "react-test-renderer";

afterEach(() => {
  cleanup();
});

test("shold render incomplete todo", () => {
  const todo = { id: 1, title: "wash chicken", completed: false };
  render(<Todo todo={todo} />);
  const todoElement = screen.getByTestId("todo-1");
  expect(todoElement).toBeInTheDocument();
  expect(todoElement).toHaveTextContent("wash chicken");
  expect(todoElement).not.toContainHTML("<strike>");
});

test("shold render complete todo", () => {
  const todo = { id: 2, title: "wash windows", completed: true };
  render(<Todo todo={todo} />);
  const todoElement = screen.getByTestId("todo-2");
  expect(todoElement).toBeInTheDocument();
  expect(todoElement).toHaveTextContent("wash windows");
  expect(todoElement).toContainHTML("</strike>");
});

test("matches snapshot", () => {
  const todo = { id: 1, title: "wash chicken", completed: false };
  const tree = renderer.create(<Todo todo={todo} />).toJSON();
  expect(tree).toMatchSnapshot();
});
