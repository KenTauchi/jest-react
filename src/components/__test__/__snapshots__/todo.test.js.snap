// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="todo-1"
>
  <h1>
    wash chicken
  </h1>
</div>
`;
